include/solvers/mpm_base.tcc

```diff
@@ -1593,18 +1593,18 @@ void mpm::MPMBase<Tdim>::initialise_nonlocal_mesh(const Json& mesh_props) {
                        .template get<double>();
 
         // Calculate beta
-        double beta = gamma / (h * h);
+        const double beta = gamma / (h * h);
         nonlocal_properties.insert(
             std::pair<std::string, double>("beta", beta));
 
         // Calculate support radius automatically
-        double r = h * std::sqrt(-std::log(tol0) / gamma);
+        const double r = std::sqrt(-std::log(tol0) / gamma) * h;
         nonlocal_properties.insert(
             std::pair<std::string, double>("support_radius", r));
 
         // Cell and node neighbourhood for LME
-        cell_neighbourhood_ = 2;
-        node_neighbourhood_ = 5;
+        cell_neighbourhood_ = static_cast<unsigned>(floor(r / h));
+        node_neighbourhood_ = 1 + 2 * cell_neighbourhood_;
```

Comment:
@migmolper This one is to compute automatically size of neighbourhood. I hope I am doing right. Will check.

Reply:
What do you think about:

node_neighbourhood_ = 1 + Ndim* max(1,cell_neighbourhood_);

Reply:
I dont think using Ndim is right. The coefficient 2 in the multiplication above means the left and right neighbors in 1D. To account for the effect of dimensions, you should power them, which will be done in the matrix assembler:
https://github.com/geomechanics/mpm/blob/268f568f0eac9dec69878cbf8e2eeb37dd1b46ea/include/linear_solvers/assemblers/assembler_base.h#L31-L54

Reply:
You are totally right. However, I think that we should check if r<h to avoid that the integer cell_neighbourhood_ becomes 0.

Reply:
But if r<h, then the function becomes only local right? Like we validate just now when gamma is large. That being said, this will yield to normal MPM and we dont have to care about nonlocal nodes at all.

Reply:
That is right, then should be convenient to use MPM instead. On the other hand, according to:

 node_neighbourhood_ = 1 + Ndim* max(1,cell_neighbourhood_);

The cases in mpm/include/linear_solvers/assemblers/assembler_base.h for node_neighbourhood_ should be: 1,3,5,..

However, you have: 0,1,2,3,4. This is for other shape functions?

Reply:
Right! that's just added for generality. I don't think we will ever use the even variants. Always odd!

Reply:
Make sense to me